'utf-8' codec can't decode byte ... in position ...: invalid continuation byte

#### Bug report

I've just installed googler from master (8253de92f8f3c7a38c6069c3e41b559066e75a74) and was working through the examples in `README.md` when I stumbled across an error. I wanted to show 5 results and autocomplete on _hello <space>_, so I typed:

`googler -d -n 5 hello\ <tab>`

There's a space between `\` and `<tab>`. Here's the result (there was no debugging output):

```
+paulo@monk:~/tmp$ googler -d -n 5 hello\ [ERROR] 'utf-8' codec can't decode byte 0xe7 in position 341: invalid continuation byte
```

Here's my environment:

**OS**: Xubuntu 17.10
**Python**: 3.6.3
**Terminal emulator**: xfce4-terminal 0.8.6 + tmux master (b5c0b2c)
**Shell**: bash 4.4-5
```
+paulo@monk:~/src/tmux$ locale
LANG=en_US.UTF-8
LANGUAGE=en_US
LC_CTYPE=pt_BR.UTF-8
LC_NUMERIC=pt_BR.UTF-8
LC_TIME=pt_BR.UTF-8
LC_COLLATE="en_US.UTF-8"
LC_MONETARY=pt_BR.UTF-8
LC_MESSAGES="en_US.UTF-8"
LC_PAPER=pt_BR.UTF-8
LC_NAME=pt_BR.UTF-8
LC_ADDRESS=pt_BR.UTF-8
LC_TELEPHONE=pt_BR.UTF-8
LC_MEASUREMENT=pt_BR.UTF-8
LC_IDENTIFICATION=pt_BR.UTF-8
LC_ALL=
```
**Bash autocompletion**: _/home/paulo/src/googler/auto-completion/bash/googler-completion.bash_

I tried using single quotes to preserve the space after _hello_ but the error was exactly the same when I tried to autocomplete after the space:

```
+paulo@monk:~/tmp$ googler -n 5 'hello [ERROR] 'utf-8' codec can't decode byte 0xe7 in position 339: invalid continuation byte
```

Not sure if this might be revelant, but the default python interpreter in Xubuntu 17.10 is 2.7.14. As I mentioned earlier, the python3 interpreter is 3.6.3. As _googler_ has the shebang `#!/usr/bin/env python3`, I imagine that shouldn't be a problem.

Let me know how I can help debug this further.

Hey, that was speed-lightning quick, thanks !

Here we go, locale set to `en_US.UTF-8`:

```
+paulo@monk:~$ locale
LANG=en_US.UTF-8
LANGUAGE=en_US
LC_CTYPE="en_US.UTF-8"
LC_NUMERIC="en_US.UTF-8"
LC_TIME="en_US.UTF-8"
LC_COLLATE="en_US.UTF-8"
LC_MONETARY="en_US.UTF-8"
LC_MESSAGES="en_US.UTF-8"
LC_PAPER="en_US.UTF-8"
LC_NAME="en_US.UTF-8"
LC_ADDRESS="en_US.UTF-8"
LC_TELEPHONE="en_US.UTF-8"
LC_MEASUREMENT="en_US.UTF-8"
LC_IDENTIFICATION="en_US.UTF-8"
LC_ALL=en_US.UTF-8
```

But the error was precisely the same:

```
+paulo@monk:~$ googler -n 5 hello\ [ERROR] 'utf-8' codec can't decode byte 0xe7 in position 341: invalid continuation byte
```

Just to clarify, I typed **hello\\** then the space key, and then the tab key to generate the autocompletions.

Just in case tmux might be getting in the way, I tried the same on a xfce4-terminal tab not running tmux, but the error was still the same:

```
+paulo@monk:~$ googler -n 5 hello\ [ERROR] 'utf-8' codec can't decode byte 0xe7 in position 335: invalid continuation byte
```

Now, this is puzzling: I changed that single line in the code (`respobj = ...`), and when I try to complete, I get the same error:

```
+paulo@monk:~/src/googler$ ./googler 'hello [ERROR] 'utf-8' codec can't decode byte 0xe7 in position 333: invalid continuation byte
```

but, if I generate the completions as @zmwangx suggested:

```
:paulo@monk:~/src/googler$ ./googler --debug --complete 'hello '
[DEBUG] googler version 3.5
[DEBUG] Python version 3.6.3
hello neighbor
hello kitty
hello darkness my old friend
hello google
hello adele
hello moto
hello tradução
hello my twenties
hello hello
hello letra
```

When I think I got it, it just slips through my fingers !

--- a/googler
+++ b/googler
@@ -2391,7 +2391,9 @@ def completer_fetch_completions(prefix):
                urllib.parse.quote(prefix, safe=''))
     # A timeout of 3 seconds seems to be overly generous already.
     resp = urllib.request.urlopen(api_url, timeout=3)
-    respobj = json.loads(resp.read().decode('utf-8'))
+    charset = resp.headers.get_content_charset()
+    logger.debug('Completions charset: %s', charset)
+    respobj = json.loads(resp.read().decode(charset))
 
     # The response object, once parsed as JSON, should look like
     #

